inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+from typing import List, Literal, Optional, Type, Union
+
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field, PositiveInt
+
+from inference.core.entities.requests.sam2 import (
+    Box,
+    Sam2Prompt,
+    Sam2PromptSet,
+    Sam2SegmentationRequest,
+)
+from inference.core.entities.responses.inference import (
+    InferenceResponseImage,
+    InstanceSegmentationInferenceResponse,
+    InstanceSegmentationPrediction,
+    Point,
+)
+from inference.core.managers.base import ModelManager
+from inference.core.utils.postprocess import masks2poly
+from inference.core.workflows.core_steps.common.entities import StepExecutionMode
+from inference.core.workflows.core_steps.common.utils import (
+    attach_parents_coordinates_to_batch_of_sv_detections,
+    attach_prediction_type_info_to_sv_detections_batch,
+    convert_inference_detections_batch_to_sv_detections,
+    load_core_model,
+)
+from inference.core.workflows.execution_engine.entities.base import (
+    Batch,
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+    BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+    STRING_KIND,
+    ImageInputField,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LONG_DESCRIPTION = """
+Run Segment Anything 2 Model
+"""
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Segment Anything 2 Model",
+            "version": "v1",
+            "short_description": "Segment Anything 2",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        },
+        protected_namespaces=(),
+    )
+    type: Literal["roboflow_core/segment_anything@v1", "SegmentAnything2Model"]
+
+    images: Union[WorkflowImageSelector, StepOutputImageSelector] = ImageInputField
+
+    boxes: StepOutputSelector(
+        kind=[
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(  # type: ignore
+        description="Boxes (from other model predictions)",
+        examples=["$steps.object_detection_model.predictions"],
+        default=None,
+    )
+
+    sam2_model: Union[
```

Comment:
I vote for name `version` or `model_version` or `model_variant` - we use `version` for YoloWorld